osu.Game/Screens/Edit/GameplayTest/EditorPlayer.cs

```diff
@@ -173,27 +173,53 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
         {
         }
 
-        private void toggleAutoplay()
+        private double previousAutoTime = 0;
+
+        private void setupAutoplay()
         {
-            if (DrawableRuleset.ReplayScore == null)
-            {
-                var autoplay = Ruleset.Value.CreateInstance().GetAutoplayMod();
-                if (autoplay == null)
-                    return;
+            var autoplay = Ruleset.Value.CreateInstance().GetAutoplayMod();
+            if (autoplay == null)
+                return;
+
+            // stores time when autoplay starts
+            previousAutoTime = GameplayClockContainer.CurrentTime;
 
-                var score = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, [autoplay]);
+            var score = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, new[] { autoplay });
 
-                // remove past frames to prevent replay frame handler from seeking back to start in an attempt to play back the entirety of the replay.
-                score.Replay.Frames.RemoveAll(f => f.Time <= GameplayClockContainer.CurrentTime);
+            // remove past frames to prevent replay frame handler from seeking back to start in an attempt to play back the entirety of the replay.
+            score.Replay.Frames.RemoveAll(f => f.Time <= GameplayClockContainer.CurrentTime);
```

Comment:
Rather than creating the score anew every time, maybe create it once at the start, and only slice the replay on demand? As in, you shouldn't be regenerating autoplay every frame, just at most taking the frames that are relevant after someone seeked back past the starting point.

Reply:
understood, only problem im having is a lack of knowledge. i dont understand how to retrieve the created score,
```csharp
// here i am stuck on where to retrieve score, so i am just going to create it just to see if it works
var autoplay = Ruleset.Value.CreateInstance().GetAutoplayMod(); // here i am stuck on where to retrieve score, so i am just going to create it
if (autoplay == null)
    return;

// end of score
var allFrames = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, new[] { autoplay }).Replay.Frames;
var drawnFrames = DrawableRuleset.ReplayScore;

// find the frames from the original replay data that should be present based on the current time
var missingFrames = allFrames.Where(f => f.Time >= GameplayClockContainer.CurrentTime && f.Time <= previousAutoTime).ToList();
previousAutoTime = GameplayClockContainer.CurrentTime;

// add the missing frames back
drawnFrames.Replay.Frames.AddRange(missingFrames);

DrawableRuleset.SetReplayScore(drawnFrames);
```